src/main/java/com/conveyal/r5/transit/TransportNetwork.java

```diff
@@ -285,25 +282,16 @@ public static InputFileType forFile(File file) {
     }
 
     /**
-     * @return an efficient implicit grid PointSet for this TransportNetwork.
+     * Build an efficient implicit grid PointSet for this TransportNetwork if it doesn't already exist. Then link that
+     * grid pointset to the street layer. This is called when a network is built for analysis purposes, and also after a
+     * scenario is applied to rebuild the grid pointset on the scenario copy of the network.
      */
-    public WebMercatorGridPointSet getGridPointSet() {
-        if (this.gridPointSet == null) {
-            synchronized (this) {
-                if (this.gridPointSet == null) {
-                    this.gridPointSet = new WebMercatorGridPointSet(this);
-                }
-            }
+    public void rebuildLinkedGridPointSet() {
+        if (gridPointSet == null) {
+            gridPointSet = new WebMercatorGridPointSet(this);
```

Comment:
It is called only in places that should already be synchronized: Scenario#applyToTransportNetwork and TransportNetworkCache#buildNetwork. The grid is usually built before the graph is serialized, and serialized along with the graph.